### DISCUSSION TITLE
Using persistent layouts with composition api and <script setup> tag

### DISCUSSION THREAD

**Discussion by rrrrando** [inertiajs/inertia#481] [Help (Vue)] [answered]
Hi!

I'm playing around with the new `<script setup>` tag in composition api. But i can't figure out how to create a persistent layout. Has anyone got it working?

**Comment by rrrrando** [comment, marked as answer]
I read the RFC and figured it out. The solution is to just add a normal script tag like so :

```vue
//Everything in here will be exposed to template
<script setup>
import Component from './ExampleCompoenent':

const item = "Some Vue stuff";
</script>
// Add a second script tag and export the layout
<script>
export default { layout: BlogLayout }
</script>

<template>
   <p>Some content here! {{ item }}</p>
</template>
```

**Comment by Fuad2Dev** [comment]
If you're using Vue 2.7 or Vue 3, you can alternatively use the [defineOptions plugin](https://vue-macros.sxzz.moe/macros/define-options.html) to define a layout within <script setup>:
```js
<script setup>
import Layout from './Layout'

defineOptions({ layout: Layout })
</script>
